### PR TITLE
Add credits field to external activities [#130192661]

### PR DESCRIPTION
**Pull request by dougmartin**

### CODE REVIEW

File: app/policies/external_activity_policy.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def set_private_before_matedit?
   end
 
   def copy?
-    user.present?
+    not_anonymous?
```
Comment (dougmartin) [inline app/policies/external_activity_policy.rb]: This was a fix for a past build break - the spec says copy can happen for non-anonymous users only.

Comment (scytacki) [inline app/policies/external_activity_policy.rb]: Perhaps this isn't consistent but, generally we pass `current_user` to the policies not `current_visitor`. `current_user` will be nil if the user is anonymous.   I believe this is the main place this is done: https://github.com/concord-consortium/rigse/blob/master/app/controllers/application_controller.rb#L30

Comment (scytacki) [inline app/policies/external_activity_policy.rb]: So the better fix would be to change this line of the spec: https://github.com/concord-consortium/rigse/blob/master/spec/policies/external_activity_policy_spec.rb#L5 to be `let(:active_user)       { nil          }`  
That might break other specs, which we should then fix.

Comment (scytacki) [inline app/policies/external_activity_policy.rb]: Also makes sense to change this one: https://github.com/concord-consortium/rigse/blob/master/spec/policies/user_policy_spec.rb#L5 So we are always using the correct pattern.

Comment (dougmartin) [inline app/policies/external_activity_policy.rb]: Both no user and the anonymous is handled with the `not_anonymous?` policy helper so I think we are safe as is:

```  
def not_anonymous?
  user && !user.anonymous?
end
```

Comment (scytacki) [inline app/policies/external_activity_policy.rb]: For completeness, we sometimes use the `policy(model)` helper in controllers. In that case `current_user` is also used. It is implicitly used by the helper provided by Pundit.

**Comment by dougmartin** [comment]
@scytacki I made the user change in the spec.

File: app/policies/material_shared_policy.rb [hunk 2]
```diff
@@ -9,6 +9,11 @@ def edit_settings?
     admin? || user && (user.admin_for_projects & record.projects).length > 0
   end
 
+  def edit_credits?
+    # Admin or admin of any project.
+    admin? || project_admin?
```
Comment (scytacki) [inline app/policies/material_shared_policy.rb]: This should only allow the project admin of the current record. This line is also doing that: https://github.com/concord-consortium/rigse/blob/master/app/policies/material_shared_policy.rb#L9 So it seems like a method named `project_admin_of_material` should be created. So the logic can be shared.

**Comment by dougmartin** [comment]
@scytacki Updated again...

**Pull request merged**